### DISCUSSION TITLE
Radio type, security type and encryption type support for System.Devices.Wifi

### DISCUSSION THREAD

**Discussion by hansmbakker** [nanoframework/Home#1201] [Q&A]
### Description

Without manually setting the network properties via the device properties dialog in visual studio, the Improv sample did not work for me.

### How to solve the problem

I believe the scan results contain more information, at least about the security type and encryption type, which could possibly be used to automatically save it in `Wireless80211Configuration` upon finding the correct network?

https://github.com/nanoframework/System.Device.Wifi/blob/598a3903245d91a6412ff1bd225c555f0ca00e70/System.Device.Wifi/WifiAdapter.cs#L107-L114

In the lines I mentioned, there is a comment saying that certain types are needed to be defined but these could use the types already defined in in https://github.com/nanoframework/System.Net/tree/main/nanoFramework.System.Net/NetworkInformation?

**Comment by josesimoes** [comment]
Welcome to nanoFrameowrk @hansmbakker!
Are you on our Discord server? It would be much easier to keep a conversation about this matter. 😉 

In any case, not entirelly sure what you're proposing? That there is a "kind-of" auto config to connect to am WiFi network?

**Reply by hansmbakker** [reply]
Hi @josesimoes!
I'm rarely on discord but will pay a visit soon to your server :)

> That there is a "kind-of" auto config to connect to am WiFi network?

Yes - I got the impression that the Improv sample only started working when I filled out all fields of the connection properties dialog, and having those set automatically based on inspection of the network would be helpful.

**Reply by josesimoes** [reply]
Couple of thoughts/questions:

1. You've proabably set the mode to connect (or active) which is oposed to disabled which would prevent WiFi from working.
2. What exactly is the "Improv sample" that you mention?
3. As minimum, if you're connecting to an auth wifi, you need SSID and pass, which you don't have when scanning.
4. When scanning, say you get N networks, how would you pick which network to connect to? And what about the password, how to supply that to the code?

Have you looked into the network helper class? https://github.com/nanoframework/System.Device.Wifi#wifinetworkhelper-usage
